### PR TITLE
feat: add User model

### PR DESCRIPTION
**Pull request by Rhotimee**
Add User model to test data

### CODE REVIEW

File: models/user/package.json [hunk 1]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@commercetools-test-data/state",
+  "version": "6.11.0",
+  "description": "Data model for commercetools API State",
+  "bugs": "https://github.com/commercetools/test-data/issues",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/commercetools/test-data.git",
+    "directory": "models/state"
```
Comment (emmenko) [inline models/user/package.json:9]:
```suggestion
    "directory": "models/user"
```

File: models/user/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    "javascript",
+    "typescript",
+    "test-data"
+  ],
+  "license": "MIT",
+  "publishConfig": {
+    "access": "public"
+  },
+  "main": "dist/commercetools-test-data-state.cjs.js",
+  "module": "dist/commercetools-test-data-state.esm.js",
```
Comment (emmenko) [inline models/user/package.json:21]:
```suggestion
  "main": "dist/commercetools-test-data-user.cjs.js",
  "module": "dist/commercetools-test-data-user.esm.js",
```

File: models/user/src/mc-user/types.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  email?: string;
+  name?: string;
+  additionalClaims?: string;
+};
+
+export type TIdTokenUserInfoGraphql = TIdTokenUserInfo & {
+  __typename: 'IdTokenUserInfo';
+};
+
+export type TMcUser = {
```
Comment (emmenko) [inline models/user/src/mc-user/types.ts:26]: Nit: would be nice to use the graphql generated types (codegen). Can be also a follow up

File: models/user/src/mc-user/types.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+export type TMcUserGraphql = TMcUser & {
+  __typename: 'User';
+};
+
+export type TMcUserDraft = {
+  email: string;
+  firstName: string;
+  lastName: string;
+};
```
[the remark this answers is not in the document]
Comment (emmenko) [reply models/user/src/mc-user/types.ts:59]: Kacper is right, the graphql mutation input has that format.

File: models/user/src/mc-user/id-token-user-info/generator.ts [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { fake, Generator } from '@commercetools-test-data/core';
+import { createRelatedDates } from '@commercetools-test-data/utils';
+import type { TIdTokenUserInfo } from '../types';
+
+const [_, getNewerDate] = createRelatedDates();
+
+const generator = Generator<TIdTokenUserInfo>({
+  fields: {
+    iss: fake((f) => f.internet.url()),
+    sub: fake((f) => f.internet.email()),
```
Comment (emmenko) [inline models/user/src/mc-user/id-token-user-info/generator.ts:10]: Nit: this is usually a UUID

File: models/user/src/mc-user/id-token-user-info/generator.ts [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    aud: fake((f) => f.internet.url()),
+    exp: fake(getNewerDate),
+    iat: fake(getNewerDate),
+    email: fake((f) => f.internet.email()),
+    name: fake((f) => f.person.firstName()),
+    additionalClaims: fake((f) => f.lorem.words()),
```
Comment (emmenko) [inline models/user/src/mc-user/id-token-user-info/generator.ts:16]: Let's leave it `null` as default

```suggestion
    additionalClaims: null,
```

File: models/user/src/mc-user/generator.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    launchdarklyTrackingTenant: fake((f) => f.lorem.word()),
+    launchdarklyTrackingCloudEnvironment: fake((f) => f.lorem.word()),
+    gravatarHash: fake((f) => f.lorem.word()),
+    defaultProjectKey: fake((f) => f.lorem.word()),
+    projects: fake(() => ({
+      count: 1,
+      offset: 0,
+      total: 1,
+      results: [McProject.random().build()],
+    })),
```
Comment (emmenko) [inline models/user/src/mc-user/generator.ts:34]: I would suggest to define this as `null` and build the model in the transformers.

**Review by emmenko — COMMENTED** [review]
🙌 Thanks a lot for working on this. Overall looks good, just some minor things to adjust.

File: models/user/package.json [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    "@babel/runtime": "^7.17.9",
+    "@babel/runtime-corejs3": "^7.17.9",
+    "@commercetools-test-data/commons": "6.11.0",
+    "@commercetools-test-data/core": "6.11.0",
+    "@commercetools-test-data/project": "workspace:^",
+    "@commercetools-test-data/utils": "6.11.0",
+    "@commercetools/platform-sdk": "^7.0.0",
+    "@faker-js/faker": "^8.0.0",
+    "add": "^2.0.6",
+    "pnpm": "^8.15.3"
```
[the remark this answers is not in the document]
Comment (Rhotimee) [reply models/user/package.json]: updated [7aba149](https://github.com/commercetools/test-data/pull/466/commits/7aba149df45fa28a1515e62ec079ece60a1c1763)

File: models/user/src/mc-user/types.ts (hunk 3 above)
Comment (Rhotimee) [reply models/user/src/mc-user/types.ts]: Yep, generated both core and mc [acf9a71](https://github.com/commercetools/test-data/pull/466/commits/acf9a715fb4684c21030b3916fe0cfbbe2af714f), [32e29bc](https://github.com/commercetools/test-data/pull/466/commits/32e29bc0354cb2c9fe6144c2f029a3dd45f5b6cb)

File: models/user/src/mc-user/id-token-user-info/generator.ts (hunk 5 above)
Comment (Rhotimee) [reply models/user/src/mc-user/id-token-user-info/generator.ts]: updated [32e29bc](https://github.com/commercetools/test-data/pull/466/commits/32e29bc0354cb2c9fe6144c2f029a3dd45f5b6cb)

File: models/user/src/mc-user/id-token-user-info/generator.ts (hunk 6 above)
Comment (Rhotimee) [reply models/user/src/mc-user/id-token-user-info/generator.ts]: updated [32e29bc](https://github.com/commercetools/test-data/pull/466/commits/32e29bc0354cb2c9fe6144c2f029a3dd45f5b6cb)

File: models/user/src/mc-user/generator.ts (hunk 7 above)
Comment (Rhotimee) [reply models/user/src/mc-user/generator.ts]: updated [32e29bc](https://github.com/commercetools/test-data/pull/466/commits/32e29bc0354cb2c9fe6144c2f029a3dd45f5b6cb)

File: models/user/src/mc-user/mc-user-draft/generator.ts [hunk 9]
```diff
@@ -0,0 +1,14 @@
+import { fake, Generator } from '@commercetools-test-data/core';
+import { fakeBusinessRoles } from '../../constants';
+import type { TMcUserDraft } from '../types';
+
+const generator = Generator<TMcUserDraft>({
+  fields: {
+    email: fake((f) => f.internet.email()),
```
Comment (kark) [inline models/user/src/mc-user/mc-user-draft/generator.ts:7]: Thank you for the updates!
I think `email` should be replaced with `password` https://github.com/commercetools/test-data/blob/448a17cdb0119faa43321bab6a0620fa0035a3c9/graphql-types/src/generated/mc.ts#L628-L633

File: models/user/src/user/types.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+type BusinessRole =
+  | 'ExecutiveManagement'
+  | 'CustomerService'
+  | 'Marketing'
+  | 'SalesAndECommerceManager'
+  | 'ProductProjectManagerOrOwner'
+  | 'Architect'
+  | 'Engineer'
+  | 'Other';
```
Comment (kark) [inline models/user/src/user/types.ts:12]: nit: just wondering, couldn't we get this type from the generated types?

File: models/user/src/user/types.ts [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+export type TUserGraphql = TCoreUser;
+
+export type TUserDraft = {
+  email: string;
+  password?: string;
+  firstName: string;
+  lastName: string;
+  language: string;
+  numberFormat?: string;
+  businessRole: BusinessRole;
```
Comment (kark) [inline models/user/src/user/types.ts:25]: nit:
```suggestion
  businessRole?: BusinessRole;
```

**Review by kark — APPROVED** [review]
Thanks 🙌 looks good to me

File: models/user/src/user/types.ts (hunk 10 above)
Comment (Rhotimee) [reply models/user/src/user/types.ts]: Yes, we can. Thanks. [f458f92](https://github.com/commercetools/test-data/pull/466/commits/f458f92496e67a42bc6e90814435d17f7f575693)

File: models/user/src/user/types.ts (hunk 11 above)
Comment (Rhotimee) [reply models/user/src/user/types.ts]: updated [f458f92](https://github.com/commercetools/test-data/pull/466/commits/f458f92496e67a42bc6e90814435d17f7f575693)

**Review by emmenko — APPROVED** [review]
Nice work, thanks a lot! 🙌

**Pull request merged**